### PR TITLE
Create React base codes

### LINKED ISSUE #492: Create React base codes
- [ ] #493

### PR DESCRIPTION
**Pull request by jinmayamashita**
## What I did

close #492 

- [x] create base code with the `pnpm create vite --template react-ts`
- [x] introduce msw for mocking data
- [x] introduce React query for auto-managed API queries
- [x] introduce ESLint

## How to test

- [x] Is this testable with jest or e2e?
- [ ] Does this need an update to the documentation?

### CODE REVIEW

**Comment by G-N555** [comment]
👀

File: examples/react/base/src/routes.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { Layout } from "@/ui/pages/layout";
+
+const Home = lazy(() => import("@/ui/pages"));
+const About = lazy(() => import("@/ui/pages/about"));
+const NotFound = lazy(() => import("@/ui/pages/not-found"));
+
+function RouteWithLayout({ children }: PropsWithChildren<{ path: string }>) {
+  return <Layout>{children}</Layout>;
+}
+
```
Comment (G-N555) [inline examples/react/base/src/routes.tsx:12]: Is it too much if we have RouteWithLayout.tsx?

I mean, I thought we could do like below.

src
|__routes
|____index.tsx
|____RouteWithLayout(Or RouteWrapper)
|______index.tsx

File: examples/react/base/src/routes.tsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import { lazy, PropsWithChildren } from "react";
+import { Switch, Route } from "wouter";
+import { Layout } from "@/ui/pages/layout";
+
+const Home = lazy(() => import("@/ui/pages"));
+const About = lazy(() => import("@/ui/pages/about"));
+const NotFound = lazy(() => import("@/ui/pages/not-found"));
+
+function RouteWithLayout({ children }: PropsWithChildren<{ path: string }>) {
```
Comment (G-N555) [inline examples/react/base/src/routes.tsx:9]: Is it too much if we separate RouteWithLayout to another directly?

I mean something like below.

src
|__Routes
|____index.tsx
|____RouteWithLayout(Or RouteWrapper)
|______index.tsx

I feel looks consistent if we have directly by each single component instead have multi components inside 1 directly.

File: examples/react/base/src/ui/nav-link.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { Link } from "wouter";
+
+export function NavLink() {
+  return (
+    <div style={{ display: "flex", gap: "8px" }}>
```
Comment (G-N555) [inline examples/react/base/src/ui/nav-link.tsx:5]: Maybe we should think about how to set styling sometimes later.
Don't think people who use boiler-plate set inline style.

**Review by G-N555 — COMMENTED** [review]
Left some comments, please take a look whenever you have time.

**Comment by ptrkdan** [comment]
(As discussed online with @jinmayamashita yesterday)

I tried running the `create-codes` command locally, but ran into a couple of issues. I am wondering if anyone else has encountered the following issues, and if they were able to resolve it.

- Appropriate `tsconfig.json` and `vite.config.js` were not included.
  - I manually added the config files from the [SolidJS base template folder](https://github.com/monstar-lab-oss/reactjs-boilerplate/tree/463a893fc97f9dbe503562462b46d361da4a4cbb/examples/solid/base), but still had the following issues.

- CSS module imports were not detected.
  - Specifically, from the generated project's `Button.tsx`.
  - Adding `"types": ["vite/client"]` to `tsconfig.json` removes the error.

- Upon running my local dev application, I get a `TypeError: _tmpl$ is not a function` error, and nothing renders on the browser.
  - I'm guessing that it's a JSX compilation issue, but have not yet determined the cause.

**Review by ptrkdan — APPROVED** [review]
LGTM!

**Pull request merged**